floors/validate_test.go

```diff
@@ -0,0 +1,291 @@
+package floors
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+func TestValidateFloorParams(t *testing.T) {
+
+	tt := []struct {
+		name     string
+		floorExt *openrtb_ext.PriceFloorRules
+		Err      string
+	}{
+		{
+			name: "Valid Skip Rate",
+			floorExt: &openrtb_ext.PriceFloorRules{Data: &openrtb_ext.PriceFloorData{
+				ModelGroups: []openrtb_ext.PriceFloorModelGroup{{
+					ModelVersion: "Version 1",
+					Schema:       openrtb_ext.PriceFloorSchema{Fields: []string{"mediaType", "size", "domain"}, Delimiter: "|"},
+					Values: map[string]float64{
+						"banner|300x250|www.website.com":              1.01,
+						"banner|300x250|*":                            2.01,
+						"banner|300x600|www.website.com|www.test.com": 3.01,
+						"banner|300x600|*":                            4.01,
+					}, Default: 0.01},
+				}}},
+			Err: "",
+		},
+		{
+			name:     "Invalid Skip Rate at Root level",
+			floorExt: &openrtb_ext.PriceFloorRules{SkipRate: -10},
+			Err:      "Invalid SkipRate = '-10' at ext.floors.skiprate",
+		},
+		{
+			name: "Invalid Skip Rate at Date level",
+			floorExt: &openrtb_ext.PriceFloorRules{Data: &openrtb_ext.PriceFloorData{
+				SkipRate: -10,
+				ModelGroups: []openrtb_ext.PriceFloorModelGroup{{
+					ModelVersion: "Version 1",
+					Schema:       openrtb_ext.PriceFloorSchema{Fields: []string{"mediaType", "size", "domain"}},
+					Values: map[string]float64{
+						"*|*|www.website.com": 15.01,
+						"*|*|*":               16.01,
+					}, Default: 0.01},
+				}}},
+			Err: "Invalid SkipRate = '-10' at  at ext.floors.data.skiprate",
+		},
+		{
+			name:     "Invalid FloorMin ",
+			floorExt: &openrtb_ext.PriceFloorRules{FloorMin: -10},
+			Err:      "Invalid FloorMin = '-10', value should be >= 0",
+		},
+		{
+			name: "Invalid FloorSchemaVersion ",
+			floorExt: &openrtb_ext.PriceFloorRules{Data: &openrtb_ext.PriceFloorData{
+				FloorsSchemaVersion: "1",
+				ModelGroups: []openrtb_ext.PriceFloorModelGroup{{
+					ModelVersion: "Version 1",
+
+					Schema: openrtb_ext.PriceFloorSchema{Fields: []string{"mediaType", "size", "domain"}, Delimiter: "|"},
+					Values: map[string]float64{
+						"banner|300x250|www.website.com": 1.01,
+						"banner|300x600|*":               4.01,
+					}, Default: 0.01},
+				}}},
+			Err: "Invalid FloorsSchemaVersion = '1', supported version 2",
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			if actErr := validateFloorParams(tc.floorExt); actErr != nil {
+				if !reflect.DeepEqual(actErr.Error(), tc.Err) {
```

Comment:
Consider using `require.Equal()` here instead of the if statement. The testify framework generates nice error messages so your t.Errorf won't be necessary.